airflow/contrib/hooks/gcs_hook.py

```diff
@@ -655,6 +655,54 @@ def insert_object_acl(self, bucket, object_name, entity, role, generation,
                 'Object ACL entry creation failed. Error was: {}'.format(ex.content)
             )
 
+    def compose(self, bucket, source_objects, destination_object):
+        """
+        Composes a list of existing object into a new object in the same storage bucket
+
+        Currently it only supports up to 32 objects that can be concatenated
+        in a single operation
+
+        https://cloud.google.com/storage/docs/json_api/v1/objects/compose
+
+        :param bucket: The name of the bucket containing the source objects.
+            This is also the same bucket to store the composed destination object.
+        :type bucket: str
+        :param source_objects: The list of source objects that will be composed
+            into a single object.
+        :type source_objects: list
+        :param destination_object: The path of the object if given.
+        :type destination_object: str
+        """
+
+        if not source_objects or not len(source_objects):
+            raise ValueError('source_objects cannot be empty.')
```

Comment:
```suggestion
            raise AirflowException('source_objects cannot be empty.')
```
I'm not sure, but I think it will be better.

Reply:
I am not sure about this. I can see it's quite common to raise ValueError for this kind of exceptions in Airflow.